### PR TITLE
[management] config

### PR DESCRIPTION
**Pull request by davidiw**
Add a config for management purposes. Lots of goodies here:
* new config file
* eliminate direct dependence on secure storage
* cleanup some code here and there
* unify everything to leverage the config

### CODE REVIEW

File: config/management/src/validator_config.rs [hunk 1]
```diff
@@ -38,10 +40,13 @@ impl ValidatorConfig {
...
     ) -> Result<Transaction, Error> {
-        let mut storage = StorageWrapper::new(
-            self.validator_backend.name(),
-            &self.validator_backend.validator_backend,
-        )?;
+        let config = self
+            .config
+            .load()?
+            .override_chain_id(self.chain_id)
+            .override_validator_backend(&self.validator_backend.validator_backend)?;
```
Comment (gregnazario) [inline config/management/src/validator_config.rs:47]: Should we do some smart checks and say if you have the ChainId with the config path to reject it?

File: config/management/src/storage.rs [hunk 2]
```diff
@@ -13,37 +13,11 @@ use std::str::FromStr;
 
 /// A helper to handle common error handling and functionality for tooling
 pub struct StorageWrapper {
-    storage_name: &'static str,
-    storage: Storage,
+    pub(crate) storage_name: &'static str,
```
Comment (gregnazario) [inline config/management/src/storage.rs:16]: Interesting, that's a new one for me. It's like I guess a Java `protected`.  We should probably be doing that lots of places in our code.

File: config/management/genesis/src/verify.rs [hunk 3]
```diff
@@ -147,45 +147,35 @@ fn compare_genesis(
...
     let expected_validator_key = validator_config.validator_network_identity_public_key;
     write_assert(
         buffer,
         VALIDATOR_NETWORK_KEY,
         actual_validator_key == expected_validator_key,
     );
 
-    let actual_fullnode_key = x25519_from_storage(FULLNODE_NETWORK_KEY, storage)
-        .map_err(|e| Error::StorageReadError(storage_name, FULLNODE_NETWORK_KEY, e))?;
+    let actual_fullnode_key = storage.x25519_public_from_private(FULLNODE_NETWORK_KEY)?;
```
Comment (gregnazario) [inline config/management/genesis/src/verify.rs:178]: Yay! less error handling, this was in my list of refactors I was going to get back to :)

File: config/management/genesis/src/validator_operator.rs [hunk 4]
```diff
@@ -3,65 +3,40 @@
...
-                Error::StorageWriteError(storage_name, constants::VALIDATOR_OPERATOR, e.to_string())
-            })?;
+        // Upload the operator name to shared storage
+        let mut shared_storage = config.shared_backend();
+        shared_storage.set(
+            constants::VALIDATOR_OPERATOR,
+            Value::String(operator_name.clone()),
+        )?;
 
         Ok(operator_name)
```
Comment (gregnazario) [inline config/management/genesis/src/validator_operator.rs:40]: We don't need to return the `operator_name` as it was passed by the caller

File: config/management/src/validator_config.rs (hunk 1 above)
Comment (davidiw) [reply config/management/src/validator_config.rs:47]: I had thought about it, but I'm kinda okay with overrides. If folks wanna do weird stuff, it makes tool writing really a pain.

File: config/management/src/storage.rs (hunk 2 above)
Comment (davidiw) [reply config/management/src/storage.rs:16]: ahh, well looking back now, I could have probably done this in such a way to use the new function.

File: config/management/genesis/src/validator_operator.rs (hunk 4 above)
Comment (davidiw) [reply config/management/genesis/src/validator_operator.rs:40]: I think the better behavior is to just return () and print success or something, but another PR :D

File: config/management/operational/src/governance.rs [hunk 5]
```diff
@@ -14,76 +17,99 @@ use libra_types::{
 use structopt::StructOpt;
 
 #[derive(Debug, StructOpt)]
-pub struct AddValidator {
-    #[structopt(long, help = "JSON-RPC Endpoint (e.g. http://localhost:8080")]
-    host: String,
-    #[structopt(long)]
-    chain_id: ChainId,
+struct RootValidatorOperation {
```
Comment (davidiw) [inline config/management/operational/src/governance.rs:20]: @zekun000, I unified your fields to match the rest of the code.

**Pull request closed without merging**